extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -8,23 +16,99 @@ export class AmpStoryShoppingConfig extends AMP.BaseElement {
     super(element);
 
     /** @private {string} */
-    this.myText_ = TAG;
+    this.element_ = element;
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+  }
 
-    /** @private {?Element} */
-    this.container_ = null;
+  /**
+   * Keys product data to product-ids and adds them to the store service.
+   * @private
+   * @param {!JsonObject} storyConfig
+   */
+  dispatchConfig_(storyConfig) {
+    const productIDtoProduct = {};
+
+    for (const item of storyConfig['items']) {
+      productIDtoProduct[item['product-tag-id']] = item;
+    }
+
+    this.storeService_.dispatch(Action.ADD_SHOPPING_STATE, productIDtoProduct);
+
+    //TODO: add call to validate config here. See validation I2I #36412 for implementation details
```

Comment:
nit: Please follow this TODO formatting (issue in parentheses):
https://github.com/ampproject/amphtml/blob/main/3p/twitter.js#L113

nit: Capitalize "**A**dd"